code/modules/projectiles/guns/hoboguns.dm

```diff
@@ -64,36 +86,21 @@
 /obj/item/gun/ballistic/automatic/hobo/zipgun
 	name = "Zip gun (9mm)"
 	icon_state = "zipgun"
-	desc = "A clever little makeshift pistol, one of the few easily-constructed firearms that accept more rounds than it has barrels. \
-		Light, compact, and packing a surprising punch, the zip gun serves as a waster's insurance policy when doing business, \
-		small enough to whip out of a coat when someone doesn't feel like paying for your raccoon pelts. \
-		A brave, enterprising waster can stuff just about anything into the improvised clipazine, though anything more powerful \
-		than a handloaded 9mm round will run the risk of voiding the warranty on your fingers."
+	desc = "A crudely handcrafted zip gun that uses 9mm ammo."
 	item_state = "gun"
-	slot_flags = ITEM_SLOT_BELT
 	w_class = WEIGHT_CLASS_SMALL
 	mag_type = /obj/item/ammo_box/magazine/zipgun
 
 	slowdown = GUN_SLOWDOWN_PISTOL_LIGHT
 	force = GUN_MELEE_FORCE_PISTOL_LIGHT
 	weapon_weight = GUN_ONE_HAND_AKIMBO
 	draw_time = GUN_DRAW_QUICK
-	fire_delay = GUN_FIRE_DELAY_NORMAL
-	autofire_shot_delay = GUN_AUTOFIRE_DELAY_SLOW
-	burst_shot_delay = GUN_BURSTFIRE_DELAY_SLOWER
-	burst_size = 1
 	damage_multiplier = GUN_EXTRA_DAMAGE_T5
-	cock_delay = GUN_COCK_RIFLE_BASE
 	init_recoil = HANDGUN_RECOIL(2.1)
 	init_firemodes = list(
-		/datum/firemode/semi_auto/slow
-	)
-	prefered_power = CASING_POWER_LIGHT_PISTOL * CASING_POWER_MOD_SURPLUS
-	misfire_possibilities = list(
-		GUN_MISFIRE_HURTS_USER(5, 5, 15, BRUTELOSS),
-		GUN_MISFIRE_THROWS_GUN(2),
-		GUN_MISFIRE_UNLOADS_GUN(0.5, 50)
+		SEMI_AUTO_SNIPER
```

Comment:
This and the other firemodes that point to a list won't actually work. Firemodes have been reworked into using a list of firemode datums, I'm almost certain if the shootcode is given a list from the firemode list, it'll runtime or something. You can use firemode datums in `modular_coyote\eris\code\gun_firemode.dm`